sdk/communication/chat/src/main/java/com/azure/android/communication/chat/AzureCommunicationChatServiceAsyncClient.java

```diff
@@ -0,0 +1,490 @@
+        this.serviceClient.listChatThreadMembers(chatThreadId, callback);
+    }
+
+    /**
+     * Gets the members of a thread.
+     * 
+     * @param chatThreadId Thread id to get members for.
+     * @param collectionCallback the Callback that receives the response collection.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    public void listChatThreadMembersPages(String chatThreadId, final Callback<AsyncPagedDataCollection<ChatThreadMember, Page<ChatThreadMember>>> collectionCallback) {
+        this.serviceClient.listChatThreadMembersPagesAsync(chatThreadId, collectionCallback);
+    }
+
+    /**
+     * Adds thread members to a thread. If members already exist, no change occurs.
+     * 
+     * @param chatThreadId Id of the thread to add members to.
+     * @param body Thread members to be added to the thread.
+     * @param callback the Callback that receives the response.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    public void addChatThreadMembers(String chatThreadId, AddChatThreadMembersOptions body, final Callback<Void> callback) {
+        this.serviceClient.addChatThreadMembers(chatThreadId, body, callback);
+    }
+
+    /**
+     * Remove a member from a thread.
+     * 
+     * @param chatThreadId Thread id to remove the member from.
+     * @param chatMemberId Id of the thread member to remove from the thread.
+     * @param callback the Callback that receives the response.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    public void removeChatThreadMember(String chatThreadId, String chatMemberId, final Callback<Void> callback) {
+        this.serviceClient.removeChatThreadMember(chatThreadId, chatMemberId, callback);
+    }
+
+    /**
+     * Creates a chat thread.
+     * 
+     * @param body Request payload for creating a chat thread.
+     * @param callback the Callback that receives the response.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    public void createChatThread(CreateChatThreadOptions body, final Callback<MultiStatusResponse> callback) {
+        this.serviceClient.createChatThread(body, callback);
```

Comment:
It's not ideal that we are returning this class (it's a list of mapping between id - status), so it won't be easy for the client to parse the thread id out of it.

I know that it's something we can't do with auto-generated code, though :)

Reply:
that's right, API exactly map to what swagger define, so any such filtering should be on the consuming side.